addons/mass_mailing/models/mailing_contact.py

```diff
@@ -161,3 +161,14 @@ def _message_get_default_recipients(self):
             'email_cc': False}
             for r in self
         }
+
+    def action_add_to_mailing_list(self):
+        ctx = dict(self.env.context, default_contact_ids=self.ids)
```

Comment:
Same here, you should re-use the existing "action_mailing_list_add_contact" instead of creating one from scratch like explained in my other comment.